packages/walletconnect/src/index.ts

```diff
@@ -188,7 +205,8 @@ export class WalletConnect extends Connector {
       if ((error as Error).message === 'User closed modal') {
         await this.deactivate(this.treatModalCloseAsError ? (error as Error) : undefined)
       } else {
-        this.actions.reportError(error as Error)
+        this.actions.resetState()
+        throw error
```

Comment:
throw or onError here?

Reply:
i think anything in `activate` is fair game to throw, so good as is imo